add a TelnetEvent for raw messages

xxx opened the pull request:
When encoding a TelnetEvent::Message, it guarantees a \r\n ending to all of its output, but that isn't always desirable. Add a new event called TelnetEvent::RawMessage that acts like a ::Message, but does not add the \r\n.

Incoming events never decode to ::RawMessage. It's intended for encoding outgoing messages only.


I added this today because I need this behavior for my project, but I'm not sure what your thoughts are on it, since it adds a new `TelnetEvent`. I haven't added tests yet because I want to make sure we're aligned on api / need first.

xxx committed:
add a TelnetEvent for raw messages

When encoding a TelnetEvent::Message, it guarantees a \r\n ending to all of its output, but that isn't always desirable. Add a new event called TelnetEvent::RawMessage that acts like a ::Message, but does not add the \r\n.

Incoming events never decode to ::RawMessage. It's intended for encoding outgoing messages only.

robertwayne commented:
I think a RawMessage event makes sense. Will need to add some doc comments to the event types to explain the difference between the two.

xxx committed:
add some tests for encode()

xxx committed:
Note a couple of TelnetEvent differences

Add a couple of notes about the differences between TelnetEvent::Message and TelnetEvent::RawMessage

xxx committed:
run rustfmt

xxx committed:
add a test for overflow during decode

robertwayne commented:
Looks good!
